rmw_zenoh_cpp/src/rmw_subscriber.cpp

```diff
@@ -318,24 +412,34 @@ rmw_take_with_info(
   rcutils_allocator_t * allocator = &subscription_data->node_->context->options.allocator;
 
   // RETRIEVE SERIALIZED MESSAGE ===============================================
-  std::string key(subscription->topic_name);
-
-  if (subscription_data->zn_messages_.find(key)
-      == subscription_data->zn_messages_.end()) {
+  if (subscription_data->zn_message_queue_.empty()) {
```

Comment:
It looks like the lock is missing in this function?

```suggestion
  std::unique_lock<std::mutex> lock(subscription_data->message_queue_mutex_);
  if (subscription_data->zn_message_queue_.empty()) {
```

Reply:
Updated: 080ae46801acf7a62547797e287a56a75cd0d9fc

I keep forgetting to change the other take function...